src/main/java/seedu/address/model/person/Birthday.java

```diff
@@ -5,32 +5,23 @@
 
 import java.time.LocalDate;
 import java.time.format.DateTimeFormatter;
-import java.time.format.DateTimeFormatterBuilder;
 import java.time.format.DateTimeParseException;
 import java.util.Locale;
+import java.util.Optional;
 
 /**
  * Represents a Person's telegram handle in the logbook.
  * Guarantees: immutable; is valid as declared in {@link #isValidBirthday(String)}
  */
 public class Birthday {
-
+    public static final DateTimeFormatter OUTPUT_FORMATTER = DateTimeFormatter.ofPattern("dd LLLL yyyy");
+    private static final String[] VALID_FORMATS = {"dd/MM/yyyy", "dd-MM-yyyy", "yyyy-MM-dd", "yyyy/MM/dd"};
+    private static final String[] VALID_FORMATS_REGEX = {"\\d{2}/\\d{2}/\\d{4}", "\\d{2}-\\d{2}-\\d{4}",
+        "\\d{4}-\\d{2}-\\d{2}", "\\d{4}/\\d{2}/\\d{2}"};
     public static final String MESSAGE_CONSTRAINTS =
-            "Birthday should not be blank and has to be in one of the given format: "
-            + "[dd/MM/yyyy] [dd-MM-yyyy] [yyyy-MM-dd] [yyyy/MM/dd]";
+            "Birthday should not be blank and has to be in one of the given format: ["
+            + String.join("] [", VALID_FORMATS) + ']';
 
-    /*
-     * The first character of the birthday must not be a whitespace,
-     * otherwise " " (a blank string) becomes a valid input.
-     */
-    public static final DateTimeFormatter VALIDATION_FORMATTER = new DateTimeFormatterBuilder()
```

Comment:
Should we include other formats? Would that be good?

Reply:
What i did before was to allow processing of single numbers for day and months

Reply:
Ok, I will create an issue for this as an extension of the feature